libcrux-ml-kem/src/ind_cca/incremental.rs

```diff
@@ -157,6 +157,58 @@ pub mod types {
         pub(super) randomness: [u8; 32],
     }
 
+    impl<const K: usize, Vector: Operations> EncapsState<K, Vector> {
+        /// Get the number of bytes, required for the state.
+        pub const fn num_bytes() -> usize {
+            SHARED_SECRET_SIZE
+                + vec_len_bytes::<K, Vector>()
+                + PolynomialRingElement::<Vector>::num_bytes()
+                + 32
+        }
+
+        /// Get the state as bytes
+        pub fn to_bytes(self, out: &mut [u8]) {
+            debug_assert!(out.len() >= Self::num_bytes());
```

Comment:
if there's any chance that this serialization will change in the future, it might be prudent to add a "version byte" to the head of this serialization.  These serializations may be stored persistently for long periods of time on user devices, and may be processed by different versions of the libcrux library.  Should any changes occur here (including length changes, or changes in the layout of bytes or additional information), new library versions should be able to process old library serialization, and old library versions should at least be able to fail gracefully to process new serializations (return a `Result`?).  Some initial byte saying "this is serialization version 1" could aid with that.

Reply:
That's a very good question. This is really just the flat byte representation of the state and don't see a reason this should ever change.
We could add a version byte. But I don't want to make things more complicated than necessary.
I think if something here changes, we should just create a new state struct.